wlauto/workloads/multiapp/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,167 @@
+
+        final String dumpsysTag = "sendToGmail";
+        final String PACKAGE = "com.google.android.gm";
+        String outputDir = parameters.getString("output_dir", "/sdcard/wa-working");
+
+        com.arm.wlauto.uiauto.gmail.UiAutomation gmail =
+            new com.arm.wlauto.uiauto.gmail.UiAutomation();
+
+        Timer result = new Timer();
+        result.start();
+
+        initDumpsysGfxInfo(PACKAGE);
+        shareUsingApp("Gmail");
+        gmail.clearFirstRunDialogues();
+
+        if (!gmail.hasComposeView()) {
+            waitForSync();
+        }
+
+        gmail.setToField(parameters);
+        gmail.setSubjectField();
+        gmail.setComposeField();
+        gmail.clickSendButton();
+        exitDumpsysGfxInfo(PACKAGE, new File(outputDir, dumpsysTag + "_gfxInfo.log"));
+
+        result.end();
+        timingResults.put("send_to_gmail_contact", result);
+    }
+
+    private void waitForSync() throws Exception {
+
+        // After the initial share request on some devices Gmail returns back
+        // to the launching app, so we need to share the photo onces more and
+        // wait for Gmail to sync.
+        com.arm.wlauto.uiauto.googlephotos.UiAutomation googlephotos =
+            new com.arm.wlauto.uiauto.googlephotos.UiAutomation();
+
+        com.arm.wlauto.uiauto.gmail.UiAutomation gmail =
+            new com.arm.wlauto.uiauto.gmail.UiAutomation();
+
+        googlephotos.tagPhoto(1);
+        shareUsingApp("Gmail");
+        gmail.clearFirstRunDialogues();
+    }
+
+    private void logIntoSkype()  throws Exception {
+
+        final String dumpsysTag = "logInToSkype";
+        final String PACKAGE = "com.skype.raider";
+        String outputDir = parameters.getString("output_dir", "/sdcard/wa-working");
+
+        com.arm.wlauto.uiauto.skype.UiAutomation skype =
+            new com.arm.wlauto.uiauto.skype.UiAutomation();
+
+        String loginName = parameters.getString("my_id");
+        String loginPass = parameters.getString("my_pwd");
+
+        Timer result = new Timer();
+        result.start();
+
+        initDumpsysGfxInfo(PACKAGE);
+        shareUsingApp("Skype");
+        skype.handleLoginScreen(loginName, loginPass);
+        confirmAccess();
+        exitDumpsysGfxInfo(PACKAGE, new File(outputDir, dumpsysTag + "_gfxInfo.log"));
+
+        result.end();
+        timingResults.put("log_into_skype", result);
+    }
+
+    private void sendToSkype() throws Exception {
+
+        final String dumpsysTag = "sendToSkype";
+        final String PACKAGE = "com.skype.raider";
+        String outputDir = parameters.getString("output_dir", "/sdcard/wa-working");
+
+        com.arm.wlauto.uiauto.skype.UiAutomation skype =
+            new com.arm.wlauto.uiauto.skype.UiAutomation();
+
+        String contactName = parameters.getString("name").replace("_", " ");
+
+        Timer result = new Timer();
+        result.start();
+
+        initDumpsysGfxInfo(PACKAGE);
```

Comment:
The workflows through the apps are quite similar so if you are not interested in the over all timings, including the share step, then I will remove them.